components/NavButtons/Signup.tsx

```diff
@@ -22,16 +22,32 @@ export const SignupButton = (): ReactElement => {
     <>
       {user ? (
         isUserProfileRoute ? (
-          <Button variant={"outline"} onClick={signOut}>
-            {"Sign Out"}
-          </Button>
+          <button
```

Comment:
Instead of duplicating the styles here - why not just import our new Button?

Reply:
My initial thought was to use the readable button but when looking at the interface for the button it relies on a href and title being passed to the component. For this part I didn't see it necessary to make an optional interface for onClick and href. By making them optional it seems like it renders the interface useless. 

Another option was to wrap the tailwind button with a div with an onClick attribute but that looked gross.

Reply:
This would be the changes to the Button.tsx file to allow for a reusable style on the SignUP.tsx Component. 
``` ts
import React from 'react'
import Link from 'next/link'
import { Navigation } from 'components/enum/navigation'

interface ButtonProps {
  onClick?: () => void;
  href?: Navigation;
  title: string;
}
 export const Button = ({ href, title }: ButtonProps): React.ReactElement => {
  return (
    <Link href={href}>
      <button className=" h-10 py-2 px-4 border-2 font-bold text-white border-white bg-transparent flex items-center rounded-md
       hover:bg-white hover:text-black hover:shadow-NavShadow hover:cursor-pointer">
        {title}
      </button>
    </Link>  )
}
```

I have a gut feeling that I don't like this because of the removal of the type-safety due to the use of the optional statement. This also may just be a skill issue on my part too lol.

Reply:
Ahaaa, I see. We want to change the interface from ``` {
  onClick?: () => void;
  href?: Navigation;
  title: string;
}```
to 
```
 {
  onClick?: () => void;
  title: string;
} | {title: string, href: string}
```

so that you can either pass in an href or an onclick and it will work. 

You'll need to modify the button, too, so that it still works with this interface.